bundles/alpha.model/src/alpha/model/util/ISLUtil.xtend

```diff
@@ -0,0 +1,114 @@
+package alpha.model.util
+
+import fr.irisa.cairn.jnimap.isl.ISLAff
+import fr.irisa.cairn.jnimap.isl.ISLBasicMap
+import fr.irisa.cairn.jnimap.isl.ISLBasicSet
+import fr.irisa.cairn.jnimap.isl.ISLConstraint
+import fr.irisa.cairn.jnimap.isl.ISLContext
+import fr.irisa.cairn.jnimap.isl.ISLDimType
+import fr.irisa.cairn.jnimap.isl.ISLMatrix
+import fr.irisa.cairn.jnimap.isl.ISLSet
+
+import static extension alpha.model.matrix.MatrixOperations.scalarMultiplication
+import static extension alpha.model.matrix.MatrixOperations.transpose
+import static extension alpha.model.util.DomainOperations.*
+
+class ISLUtil {
+	
+	/** Creates an ISLBasicSet from a string */
+	def static toISLBasicSet(String descriptor) {
+		ISLBasicSet.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Creates an ISLSet from a string */
+	def static toISLSet(String descriptor) {
+		ISLSet.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Creates an ISLBasicMap from a string */
+	def static toISLBasicMap(String descriptor) {
+		ISLBasicMap.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Creates an ISLBasicSet from a string */
+	def static toISLAff(String descriptor) {
+		ISLAff.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Transposes an ISLMatrix */
+	def static transpose(ISLMatrix matrix) {
+		ISLMatrix.buildFromLongMatrix(matrix.toLongMatrix.transpose)
+	}
+	
+	/** Returns the integer point closest to the origin in set without parameter context */
+	def static long[] integerPointClosestToOrigin(ISLBasicSet set) {
+		set.copy.samplePoint.coordinates.subList(set.nbParams, set.nbParams + set.nbIndices)
+	}
+	
+	def static isTrivial(ISLBasicSet set) {
+		var origin = ISLBasicSet.buildUniverse(set.space.copy)
+		for (i : 0..<set.space.nbIndices) {
+			val aff = ISLAff.buildZero(set.space.copy.toLocalSpace)
+			origin = origin.addConstraint(aff.setCoefficient(ISLDimType.isl_dim_in, i, 1).toEqualityConstraint)
+		}
+		
+		set.copy.toSet.subtract(origin.toSet).isEmpty
+	}
+	
+	def static isTrivial(ISLSet set) {
```

Comment:
Consider the following rewrite of `isTrivial(ISLSet set)`, which uses a more functional style and avoids repeatedly calling the chain `set.space.copy.toLocalSpace` by computing that once and then just calling `copy`.

Also note that I added `set.copy` to the final line to make sure it doesn't get freed in case the caller still needs it.

```
val zero = ISLAff.buildZero(set.space.copy.toLocalSpace)
val constraints =
	(0 ..< set.space.nbIndices)
	.map[i | zero.copy.setCoefficient(ISLDimType.isl_dim_in, i, 1).toEqualityConstraint]
val origin = constraints.fold(ISLSet.buildUniverse(set.space.copy), [s, c | s.addConstraint(c)])

set.copy.subtract(origin).isEmpty
```

Reply:
I like it. Done.